src/float_cast.h

```diff
@@ -106,17 +106,40 @@
 
 		return intgr ;
 	}
-#elif (defined (WIN32) || defined (_WIN32)) && defined(_M_3X64)
+#elif (defined (WIN32) || defined (_WIN32)) && defined(_M_X64)
 
 	#include <math.h>
 	#include <immintrin.h>
+	#include <emmintrin.h>
 
-	__inline long int
-	lrintf (float flt)
+	#ifdef _MSC_VER
+		#pragma function(lrint, lrintf)
+	#endif
+
+	__inline 
+	long int lrint(double flt)
 	{
-		return _mm_cvt_ss2si(_mm_set_ss(flt));
+       return _mm_cvtsd_si32(_mm_set_sd(flt));
+    }
+
+	__inline 
+	long int lrintf (float flt)
+	{
+		return _mm_cvtss_si32(_mm_set_ss(flt));
 	}
 
+	__inline 
+	long long int llrint(double flt)
+    {
+        return _mm_cvtsd_si64(_mm_set_sd(flt));
```

Comment:
converts to 64 bit, so assume sizeof(long long) == 8

Reply:
Less surprising than the previous two functions, but a static_assert won't hurt either.